ITISFoundation/osparc-simcore
# ♻️ Refactors service listing queries and 🗃️ drops ``service_latest`` cache table

## What do these changes do?

This PR address some refactoring agreed with some the reviewers after  https://github.com/ITISFoundation/osparc-simcore/pull/4038 . Specifically:

- 🗃️ removes ``services_latest`` table from pg
    -  ♻️ removes step to fill ``services_latest`` in periodic background task of ``catalog`` service
- ✨ This table is replaced by a subquery  in ``simcore_postgres_database.utils_services``


## Related issue/s

- follows from https://github.com/ITISFoundation/osparc-simcore/pull/4038 
- related to ITISFoundation/osparc-issues#683
- closes https://github.com/ITISFoundation/osparc-simcore/issues/4032

## How to test

- test coverage
- **Manual test**:
   - deploy stack and open web api swagger site:
   - ``GET /v0/services`` should list latest versions of (see ``view_url`` to check version)
